crypto/threads_pthread.c

```diff
@@ -525,6 +523,9 @@ CRYPTO_RCU_LOCK *ossl_rcu_lock_new(int num_writers, OSSL_LIB_CTX *ctx)
 {
     struct rcu_lock_st *new;
 
+    if (num_writers < 0 || num_writers > INT32_MAX)
```

Comment:
This is unnecessary since the following `if (num_writers < 3)` does the right thing,
and especially the `num_writers > INT32_MAX`
is likely to cause compiler warnings about the impossible condition.